src/pages/Home/index.js

```diff
@@ -72,46 +74,157 @@ function Deals({ item }) {
   );
 }
 
+function ProductModal({ selectedId, closeModal, product }) {
+  const [cart, setCarting] = useState([]);
+  function handleAddProduct(product) {
+    setCarting((cart) => [...cart, product]);
+  }
+
+  function handleAdd() {
+    handleAddProduct(selectedId);
+  }
```

Comment:
Do we need to have these two separate functions? It seems like all `handleAdd` is doing is calling `handleAddProduct`. We should be able to make use of a single function for this. Perhaps try the following

```suggestion
  function handleAddProductToCart() {
    setCarting((cart) => [...cart, selectedId]);
  }
```

Reply:
The  new code worked fine